browser/ui/webui/settings/settings_cookies_view_handler_unittest.cc

```diff
@@ -0,0 +1,269 @@
+  }
+
+  void SetupTreeModelForTesting() {
+    mock_browsing_data_cookie_helper =
+        base::MakeRefCounted<browsing_data::MockCookieHelper>(
+            profile()->GetDefaultStoragePartition());
+    mock_browsing_data_local_storage_helper =
+        base::MakeRefCounted<browsing_data::MockLocalStorageHelper>(
+            profile()->GetDefaultStoragePartition());
+
+    auto container = std::make_unique<LocalDataContainer>(
+        mock_browsing_data_cookie_helper,
+        /*database_helper=*/nullptr, mock_browsing_data_local_storage_helper,
+        /*session_storage_helper=*/nullptr,
+        /*indexed_db_helper=*/nullptr,
+        /*file_system_helper=*/nullptr,
+        /*quota_helper=*/nullptr,
+        /*service_worker_helper=*/nullptr,
+        /*data_shared_worker_helper=*/nullptr,
+        /*cache_storage_helper=*/nullptr);
+    auto mock_cookies_tree_model = std::make_unique<CookiesTreeModel>(
+        std::move(container), profile()->GetExtensionSpecialStoragePolicy());
+
+    mock_browsing_data_local_storage_helper->AddLocalStorageForStorageKey(
+        blink::StorageKey::CreateFromStringForTesting(kTestOrigin1), 2);
+    mock_browsing_data_local_storage_helper->AddLocalStorageForStorageKey(
+        blink::StorageKey::CreateFromStringForTesting(kTestOrigin2), 3);
+
+    mock_browsing_data_cookie_helper->AddCookieSamples(GURL(kTestOrigin1),
+                                                       kTestCookie1);
+    mock_browsing_data_cookie_helper->AddCookieSamples(GURL(kTestOrigin2),
+                                                       kTestCookie2);
+
+    handler()->SetCookiesTreeModelForTesting(
+        std::move(mock_cookies_tree_model));
+  }
+
+  void NotifyTreeModel() {
+    mock_browsing_data_local_storage_helper->Notify();
+    mock_browsing_data_cookie_helper->Notify();
+  }
+
+  void SetupHandlerWithTreeModel() {
+    SetupTreeModelForTesting();
+
+    base::Value::List reload_args;
+    reload_args.Append(kCallbackId);
+    handler()->HandleReloadCookies(reload_args);
+
+    task_environment()->RunUntilIdle();
+    NotifyTreeModel();
+
+    // After batch end, the handler will have posted a task to complete
+    // the callback.
+    task_environment()->RunUntilIdle();
+    const content::TestWebUI::CallData& data = *web_ui()->call_data().back();
+    EXPECT_EQ("cr.webUIResponse", data.function_name());
+    EXPECT_EQ(kCallbackId, data.arg1()->GetString());
+    ASSERT_TRUE(data.arg2()->GetBool());
+
+    web_ui_->ClearTrackedCalls();
+  }
+
+  content::TestWebUI* web_ui() { return web_ui_.get(); }
+  CookiesViewHandler* handler() { return handler_.get(); }
+
+ private:
+  std::unique_ptr<content::TestWebUI> web_ui_;
+  std::unique_ptr<CookiesViewHandler> handler_;
+
+  // Ref pointers to storage helpers used in the tree model used for testing.
+  // Retained to allow control over batch update completion.
+  scoped_refptr<browsing_data::MockLocalStorageHelper>
```

Comment:
These variables are not named according to convention required by the the style guide [1]. I'm flagging this to raise some awareness, to authors and reviewers, as I've come across a few places today where such cases seemed to have got into master.

[1] https://google.github.io/styleguide/cppguide.html#Variable_Names